dev/benchmarking/ddl/1_tables.sql

```diff
@@ -0,0 +1,291 @@
+  IS 'The output of `sysctl -n hw.memsize`.';
+COMMENT ON COLUMN public.machine.cpu_actual_frequency_Hz
+  IS 'The output of `sysctl -n hw.cpufrequency`.';
+
+-- UNIT
+CREATE TABLE IF NOT EXISTS public.unit
+(
+  id SERIAL PRIMARY KEY
+  , unit_name citext NOT NULL UNIQUE
+  , benchmark_type_id integer NOT NULL
+  , FOREIGN KEY (benchmark_type_id) REFERENCES public.benchmark_type(id)
+);
+COMMENT ON TABLE public.unit IS 'The actual units for a reported benchmark.';
+COMMENT ON COLUMN public.unit.unit_name
+  IS 'For example: ms, ns, microseconds.';
+
+-- ENVIRONMENT
+CREATE TABLE IF NOT EXISTS public.environment
+(
+  id SERIAL
+  , benchmark_language_id integer NOT NULL
+  , language_version_id integer NOT NULL
+  , language_dependencies_id integer NOT NULL
+  , PRIMARY KEY
+      (id, benchmark_language_id, language_version_id)
+  , FOREIGN KEY
+      (benchmark_language_id)
+      REFERENCES public.benchmark_language
+  , FOREIGN KEY
+      (language_version_id, benchmark_language_id)
+      REFERENCES public.language_version(id, benchmark_language_id)
+  , FOREIGN KEY
+      (language_dependencies_id, benchmark_language_id)
+      REFERENCES public.language_dependencies(id, benchmark_language_id)
+);
+COMMENT ON TABLE public.environment
+  IS 'Identifies a build environment for a specific suite of benchmarks.';
+
+-- BENCHMARK
+CREATE TABLE IF NOT EXISTS public.benchmark
+(
+  id SERIAL
+  , benchmark_language_id integer NOT NULL
+  , benchmark_name citext NOT NULL
+  , parameter_names text[]
+  , lessisbetter boolean NOT NULL
+  , benchmark_description text NOT NULL
+  , benchmark_version citext NOT NULL
+  , unit_id integer NOT NULL
+  , PRIMARY KEY (id, benchmark_language_id)
+  , FOREIGN KEY (benchmark_language_id) REFERENCES public.benchmark_language
+  , FOREIGN KEY (unit_id) REFERENCES public.unit
+);
+COMMENT ON TABLE public.benchmark
+  IS 'Identifies an individual benchmark.';
+COMMENT ON COLUMN public.benchmark.parameter_names
+  IS 'A list of strings identifying the parameter names in the benchmark.';
+COMMENT ON COLUMN public.benchmark.lessisbetter
+  IS 'True if a smaller benchmark value is better.';
+COMMENT ON COLUMN public.benchmark.benchmark_version
+  IS 'Can be any string. In Airspeed Velocity, the version is '
+     'by default the hash of the entire code string for the benchmark.';
+
+-- BENCHMARK_RESULT
+CREATE TABLE IF NOT EXISTS public.benchmark_result
+(
+  id BIGSERIAL PRIMARY KEY
+  , parameter_values jsonb NOT NULL DEFAULT '{}'::jsonb
+  , value numeric NOT NULL
+  , git_commit_timestamp timestamp (0) with time zone NOT NULL
+  , git_hash text NOT NULL
+  , val_min numeric
+  , val_q1 numeric
+  , val_q3 numeric
+  , val_max numeric
+  , std_dev numeric NOT NULL
+  , n_obs integer NOT NULL
+  , machine_id integer NOT NULL
+  , benchmark_language_id integer NOT NULL
+  , language_version_id integer NOT NULL
+  , environment_id integer NOT NULL
+  , benchmark_id integer NOT NULL
```

Comment:
it might be useful to have a notes field on this table, so results can be annotated with other interesting information.  Also potentially a key-value pair relationship to add additional metadata (one thing that comes to mind is average load or some other measure of CPU utilization)

Reply:
Not hard to add either of these; added.